### PR TITLE
API for checking collection exists

### PR DESCRIPTION
**Pull request by gabrieldemian**
### All Submissions:

/claim #3332 

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/api/src/grpc/proto/collections.proto [hunk 1]
```diff
@@ -50,6 +50,14 @@ message GetCollectionInfoRequest {
   string collection_name = 1; // Name of the collection
 }
 
+message CollectionExistsRequest {
+  string collection_name = 1;
+}
+
+message CollectionExistsResponse {
+  bool exists = 1;
+}
```
Comment (coszio) [inline lib/api/src/grpc/proto/collections.proto:59]: Just for consistency, we should also add timing in grpc, same as we do in http

**Review by coszio — CHANGES_REQUESTED** [review]
Good work, I kinda want to approve this already, but I have this 1 item to resolve

File: lib/api/src/grpc/proto/collections.proto (hunk 1 above)
Comment (gabrieldemian) [reply lib/api/src/grpc/proto/collections.proto]: thank you, just did this

**Comment by generall** [comment]
Hi @gabrieldemian,

it looks like the REST api response is inconsistent now:

image

**Review by generall — CHANGES_REQUESTED** [review]
Please fix the REST output format.

Also I noticed that our CI script for checking read-only API list consistency is broken. Here is a fix attached
[fix-read-only.diff.txt](https://github.com/qdrant/qdrant/files/14072832/fix-read-only.diff.txt)

**Comment by gabrieldemian** [comment]
> Hi @gabrieldemian,
> 
> it looks like the REST api response is inconsistent now:
> 
> image

I fixed this inconsistency, it's working as expected on REST. On grpc, when the collection doesn't exist, `grpcurl` result is an empty object, however, the result is actually correct (I did a simple println! of the tonic::Response). I'm not sure if this is related to grpcurl.

image
image

**Comment by gabrieldemian** [comment]
> Please fix the REST output format.
> 
> Also I noticed that our CI script for checking read-only API list consistency is broken. Here is a fix attached [fix-read-only.diff.txt](https://github.com/qdrant/qdrant/files/14072832/fix-read-only.diff.txt)

fixed that, I tested both REST and grpc on my terminal and the responses were as expected, a single time value outside the result object

**Review by timvisee — APPROVED** [review]

**Review by generall — APPROVED** [review]

**Review by generall — APPROVED** [review]
LGMT

**Pull request merged**